src/components/services-view/services-view.controller.ts

```diff
@@ -21,44 +19,39 @@ export class ServicesViewController implements angular.IController {
     this.$filter = $filter;
     this.$scope = $scope;
     this.$timeout = $timeout;
-    this.ctrl.loading = true;
+    this.ctrl.loaded = false;
+    this.ctrl.isEmpty = false;
   }
 
   public $onInit() {
-    this.ctrl.allItems = [];
     this.ctrl.currentFilter = 'all';
     this.ctrl.currentSubFilter = null;
 
-    this.updateAll();
-
     this.debounceResize = _.debounce(this.resizeExpansion, 50, { maxWait: 250 });
     angular.element(window).bind('resize', this.debounceResize);
     $(window).on('resize.services', this.debounceResize);
   }
 
   public $onChanges(onChangesObj: angular.IOnChangesObject) {
-    if (onChangesObj.serviceClasses && !onChangesObj.serviceClasses.isFirstChange()) {
-      this.ctrl.serviceClasses = onChangesObj.serviceClasses.currentValue;
-      this.serviceClassesLoaded = true;
-      this.updateServiceClasses();
-    }
-
-    if (onChangesObj.imageStreams && !onChangesObj.imageStreams.isFirstChange()) {
-      this.ctrl.imageStreams = onChangesObj.imageStreams.currentValue;
-      this.imageStreamsLoaded = true;
-      this.updateImageStreams();
+    if (onChangesObj.catalogServiceItems && (!onChangesObj.catalogServiceItems.isFirstChange() ||
+      !_.isEmpty(onChangesObj.catalogServiceItems.currentValue))) {
+      this.ctrl.filteredItems = this.ctrl.catalogServiceItems;
+      this.ctrl.categories = this.catalog.removeEmptyCategories(this.ctrl.filteredItems);
+      this.ctrl.subCategories = this.getSubCategories('all');
+      this.ctrl.isEmpty = _.isEmpty( this.ctrl.catalogServiceItems);
+      this.ctrl.loaded = true;
     }
```

Comment:
yeah, I was trying to accommodate production code (which is async so $onChanges gets called twice) and unit tests (where $onChanges only is called once).
I fixed with a much simpler test.